File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProductDifference = function(nums) {
  nums.sort((a, b) => a - b)
  const l = nums.length
  return nums[l-1]*nums[l-2] - nums[0]*nums[1]
};

File: 1913-maximum-product-difference-between-two-pairs/1913-maximum-product-difference-between-two-pairs.go
func maxProductDifference(nums []int) int {
  maxIndex, minIndex := 0, 0
  for i, v := range nums {
    if v > nums[maxIndex] {
      maxIndex = i
    }
    if v < nums[minIndex] {
      minIndex = i
    }
  }
  max2Index, min2Index := minIndex, maxIndex
  for i, v := range nums {
    if i != maxIndex && v > nums[max2Index] {
      max2Index = i
    }
    if i != minIndex && v < nums[min2Index] {
      min2Index = i
    }
  }
  
  return nums[maxIndex]*nums[max2Index] - nums[minIndex]*nums[min2Index]
}